presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManagerStats.java

```diff
@@ -42,18 +43,26 @@
     private final CounterStat consumedInputBytes = new CounterStat();
     private final CounterStat consumedCpuTimeSecs = new CounterStat();
     private final TimeStat executionTime = new TimeStat(MILLISECONDS);
+    private final TimeStat queuedTime = new TimeStat(MILLISECONDS);
     private final DistributionStat wallInputBytesRate = new DistributionStat();
     private final DistributionStat cpuInputByteRate = new DistributionStat();
 
+    public void queryQueued()
+    {
+        queuedQueries.incrementAndGet();
+    }
+
     public void queryStarted()
     {
         startedQueries.update(1);
         runningQueries.incrementAndGet();
+        queuedQueries.decrementAndGet();
     }
 
     public void queryStopped()
     {
         runningQueries.decrementAndGet();
+        queuedQueries.decrementAndGet();
```

Comment:
Aren't you decrementing the `queuedQueries` twice? First the query will get queued (`queryQueued` will get called), then the query will start running (`queryStarted ` gets called and decrements), then query will finish (`queryStopped ` will get called and decrements), right?